src/SFML/Audio/SoundFileWriterOpus.cpp

```diff
@@ -0,0 +1,259 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2016 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Audio/SoundFileWriterOpus.hpp>
+#include <SFML/System/Err.hpp>
+#include <algorithm>
+#include <cstring>
+#include <ctime>
+#include <cassert>
+
+
+// Anonymous namespace
+namespace
+{
+    // Make sure to write int into buffer little endian
+    void writeUint32(unsigned char* buffer, sf::Uint32 value)
+    {
+        buffer[0] = static_cast<unsigned char>(value & 0x000000FF);
+        buffer[1] = static_cast<unsigned char>((value & 0x0000FF00) >> 8);
+        buffer[2] = static_cast<unsigned char>((value & 0x00FF0000) >> 16);
+        buffer[3] = static_cast<unsigned char>((value & 0xFF000000) >> 24);
+    }
+}
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::check(const std::string& filename)
+{
+    std::string extension = filename.substr(filename.find_last_of(".") + 1);
+    std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
```

Comment:
Adding answer to [Laurent's comment](https://github.com/SFML/SFML/pull/1077#issuecomment-204992310) here:

> He just copied the code that I wrote in other encoders. And I probably did that on purpose (I never explicitly scope a symbol just for fun). Maybe the C++  tolower  doesn't compile here due to template/overload stuff? I don't remember.

There's an additional overload of `std::tolower` in `<locale>` which takes a locale as second parameter. When this header is included, there may indeed be an ambiguity. So I guess you encountered problems because some standard libraries include that header through others... In that case you're right, the `::tolower` might be better.

...or to ensure 100% portability, `static_cast<int (*)(int)>(&std::tolower)` :stuck_out_tongue: